libraries/ms-common/src/x86/uart.c

```diff
@@ -12,30 +14,26 @@ typedef struct {
 
 static UartSocketData s_sock[NUM_UART_PORTS];
 
-void uart_receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
-                           size_t rx_len, void *context);
-
-void uart_receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
-                           size_t rx_len, void *context) {
+static void prv_receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
+                                 size_t rx_len, void *context) {
   UartSocketData *sock = context;
   sock->storage->rx_handler((uint8_t *)rx_data, rx_len, context);
 }
 
 StatusCode uart_init(UARTPort uart, UARTSettings *settings, UARTStorage *storage) {
   // create module name from UART_PORT number
-  size_t module_name_len = 6;
-  char *module_name = malloc(module_name_len);
-  snprintf(module_name, module_name_len, "%s%d", "uart", ((int)uart + 1));
+  char *module_name = malloc(UART_MODULE_NAME_LEN);
+  snprintf(module_name, UART_MODULE_NAME_LEN, "%s%d", "uart", ((int)uart + 1));
 
   s_sock[uart].port = uart;
   s_sock[uart].storage = storage;
   s_sock[uart].storage->rx_handler = settings->rx_handler;
   s_sock[uart].storage->context = settings->context;
 
-  status_ok_or_return(x86_socket_init(&s_sock[uart].thread, module_name, uart_receiver_wrapper,
-                                      (void *)&s_sock[uart]));
+  StatusCode ret_val = x86_socket_init(&s_sock[uart].thread, module_name, prv_receiver_wrapper,
+                                       (void *)&s_sock[uart]);
 
-  return STATUS_CODE_OK;
+  return ret_val;
```

Comment:
You could just return directly. There's no point in creating a variable.

You also don't need to explicitly cast the pointer to a void. It will implicitly cast it.